lib/index.js

```diff
@@ -30,9 +30,59 @@ function handleExpectedCallback(results, logAction, done, areEqual){
   };
 }
 
-module.exports = function(expected, actual, logAction, done, areEqual){
+/**
+ * Invokes two implemmentations of the same feature, it compares their results and invokes
+ * the logAction callback when results are different
+ * 
+ * @param  {object}   options This parameter should contain the following properties:
+ * 
+ *   - expected: {Function}   it takes a 'cb' that performs the operation that
+ *                            results in the expected outcome. Commonly, the current
+ *                            implementation of your feature.
+ *                            
+ *   - actual: {Function}     It takes a cb that performs the operation that
+ *                            results in the actual outcome. Commonly, the new
+ *                            implementation of your feature.
+ *                            
+ *   - logAction: {Function}  Only invoked if there is an error in expected or the results
+ *                            are different.
+ *                            
+ *   - areEqual: {Function}   (optional) A function that takes both results and must return
+ *                            a boolean indicating if both results are equal or not.    
+ * 
+ * @param  {Function} done    [It will be invoked with the result of the expected operation.
+ *                            This is commonly the callback you were using for the original
+ *                            implementation of the feature.
+ */
+module.exports = function(options, done){
+
+  if (typeof done !== 'function') {
+    throw new Error('\'done\' argument must be a function.');
+  }
+
+  if (!options || typeof options !== 'object') {
```

Comment:
need to `return`. also use https://github.com/hernanhht/immediate-invocation, caller expects asynchronous invocation for callback